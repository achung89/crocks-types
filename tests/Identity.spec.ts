/// <reference path="../templates/index.d.ts" />

import { Identity } from "crocks";

// @dts-jest:pass:snap
Identity(5);

// @dts-jest:fail:snap
Identity(5).map("whatever");

// @dts-jest:pass:snap
Identity(5).map(() => "");

// @dts-jest:fail:snap
Identity("a string").map((v) => v * 2);

// @dts-jest:pass:snap
Identity.of("Hello");

// @dts-jest:fail:snap
Identity(5).chain("")

// @dts-jest:fail:snap
Identity(5).chain(() => 10)

// @dts-jest:pass:snap
Identity(5).chain(() => Identity(5))

// @dts-jest:fail:snap
Identity(5).ap(10)

// @dts-jest:pass:snap
Identity(a => b => a + b).ap(Identity.of(10)).ap(Identity.of(10))

// @dts-jest:fail:snap
Identity(5).ap(Identity.of(10))

// @dts-jest:fail:snap
Identity.type(10) 

// @dts-jest:pass:snap
Identity.type() 

// @dts-jest:pass:snap
Identity(5).valueOf()

// @dts-jest:fail:snap
Identity(5).traverse(5, Identity)

// @dts-jest:fail:snap
Identity(5).traverse(() => 5, undefined)

// @dts-jest:pass:snap
Identity(5).sequence(Identity)

// @dts-jest:fail:snap
Identity(5).sequence(5)
