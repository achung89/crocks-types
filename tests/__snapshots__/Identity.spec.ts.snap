// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Identity("a string").map((v) => v * 2) (type) should match snapshot: Identity("a string").map((v) => v * 2) 1`] = `"The left-hand side of an arithmetic operation must be of type 'any', 'number' or an enum type."`;

exports[`Identity(5) (type) should match snapshot: Identity(5) 1`] = `"import(\\"crocks\\").IdentityInstance<number>"`;

exports[`Identity(5).ap(10) (type) should match snapshot: Identity(5).ap(10) 1`] = `"Argument of type '10' is not assignable to parameter of type 'IdentityInstance<{}>'."`;

exports[`Identity(5).ap(Identity.of(10)) (type) should match snapshot: Identity(5).ap(Identity.of(10)) 1`] = `"import(\\"crocks\\").IdentityInstance<number>"`;

exports[`Identity(5).chain("") (type) should match snapshot: Identity(5).chain("") 1`] = `"Argument of type '\\"\\"' is not assignable to parameter of type '(t: number) => IdentityInstance<{}>'."`;

exports[`Identity(5).chain(() => 10) (type) should match snapshot: Identity(5).chain(() => 10) 1`] = `
"Argument of type '() => number' is not assignable to parameter of type '(t: number) => IdentityInstance<{}>'.
  Type 'number' is not assignable to type 'IdentityInstance<{}>'."
`;

exports[`Identity(5).chain(() => Identity(5)) (type) should match snapshot: Identity(5).chain(() => Identity(5)) 1`] = `"import(\\"crocks\\").IdentityInstance<number>"`;

exports[`Identity(5).map("whatever") (type) should match snapshot: Identity(5).map("whatever") 1`] = `"Argument of type '\\"whatever\\"' is not assignable to parameter of type '(t: number) => {}'."`;

exports[`Identity(5).map(() => "") (type) should match snapshot: Identity(5).map(() => "") 1`] = `"import(\\"crocks\\").IdentityInstance<string>"`;

exports[`Identity.of("Hello") (type) should match snapshot: Identity.of("Hello") 1`] = `"import(\\"crocks\\").IdentityInstance<string>"`;

exports[`Math.max('123') (type) should match snapshot: Math.max('123') 1`] = `"Argument of type '\\"123\\"' is not assignable to parameter of type 'number'."`;

exports[`Math.max(1) (type) should match snapshot: Math.max(1) 1`] = `"number"`;
